tests/test_ncbi.py

```diff
@@ -0,0 +1,9 @@
+import pytest
+
+from semantic_search.ncbi import _medline_to_docs
+
+
+def test_invalid_uid_test():
+    with pytest.raises(TypeError):
```

Comment:
Is this correct? I thought we were trying to catch a raised `HTTPException`

Reply:
We are raising a `HTTPException` but `pytest` throws an error with that. However, when I use a `TypeError` catch, `pytest` accepts it. Not fully sure why this happens, but it happens after we made the updates right now.
This is what happens when I raise `HTTPException`:

![image](https://user-images.githubusercontent.com/57552053/115312645-af3db300-a13f-11eb-924d-f333b54b39a2.png)

Reply:
The goal of the unit test is to assert that we are raising the error we think we are, which in this case is a `HTTPException`. Remember you write tests in order to catch mistakes in your code, so you shouldn't rewrite a test just to get it to pass.

In this case I suspect the problem is with the unit test

```python
def test_invalid_uid_test():
    with pytest.raises(TypeError):
        uid = ["93846392868"]
        _medline_to_docs(uid)
```

This is passing a `List[str]` to `_medline_to_docs`, but based on the header of `_medline_to_docs`, it expects a `List[Dict[str, str]]`. You might have to examine `_medline_to_docs` a little closer in the `ncbi` module, specifically what kind of objects are passed to it, and update the unit test to pass a similar object.

Reply:
I think it should be as easy as updating the unit test to:

```python
def test_invalid_uid_test():
    with pytest.raises(HTTPException):
        uid = ["93846392868"]
        records = [{"id:": [uid]}]
        _medline_to_docs(records)
```

but do confirm and convince yourself that is correct
```

Reply:
This update does seem to work with `HTTPException`. However, I'm still a little confused because `pytest` was raising a `HTTPException` (like it was supposed to), but when we added in the `(record["id:"][-1])`, it somehow changed to a `TypeError`. There was probably something else that was causing an issue because I was always passing a `List[str]` and it seemed to worked. 
This seems to work now and I have updated it. Thanks again for the help.

Reply:
Hey @JohnGiorgi, are we good to merge?